homework-3/src/components/shared/SidebarNavigation/SidebarNavigation.tsx

```diff
@@ -0,0 +1,68 @@
+"use client";
+import { Flex, Heading } from "@chakra-ui/react";
+import NextLink from "next/link";
+import styles from "./SidebarNavigation.module.css";
+import { useRef } from "react";
+
+export default function SidebarNavigation() {
+  const navListEl = useRef<HTMLDivElement>(null);
+
+  function changeActiveLink(e: any) {
+    const links = navListEl.current?.childNodes as NodeListOf<HTMLAnchorElement>;
+    links.forEach((link: HTMLAnchorElement) =>
+      link.classList.remove(styles.active)
+    );
+    e.currentTarget.classList.add(styles.active);
+  }
+
+  return (
+    <div className={styles.navWrapper}>
+      <Flex
+        className={styles.navbar}
+        justifyContent="space-between"
+      >
+        <nav className={styles.navItems}>
+          <Heading
+            mb={10}
+            size="s"
+            whiteSpace="nowrap"
+          >
+            <i className="fa-solid fa-tv"></i> TV shows APP
+          </Heading>
+          <Flex
+            flexWrap="wrap"
+            flexDirection="column"
+            gap={3}
+            ref={navListEl}
+          >
+            <NextLink
+              href={`/shows/all-shows`}
+              onClick={(e) => changeActiveLink(e)}
+              className={`${styles.navLink} ${styles.active}`}
+            >
+              All shows
+            </NextLink>
```

Comment:
Šta misliš bi li se ovo moglo malo generičnije riješit? Hintovi - komentar ispod i Next navigation pathname
```suggestion
            <NextLink
              href={`/shows/all-shows`}
              onClick={(e) => changeActiveLink(e)}
              className={`${styles.navLink} ${styles.active}`}
            >
              All shows
            </NextLink>
            
            {/*
              save navigation items to some /date folder
              foreach data links
              check if path from current loop's element is equal to pathname
              if yes, add active class
            */}
```